src/hooks/use-assistant.ts

```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from 'react';
+
+interface Message {
+	content: string;
+	role: 'user' | 'assistant';
+}
+
+export const useAssistant = ( selectedSiteId: string ) => {
+	const [ messages, setMessages ] = useState< Message[] >( [] );
+
+	useEffect( () => {
+		const storedChat = localStorage.getItem( selectedSiteId );
```

Comment:
The user id could be added here as an extra info to the data key to make chats unique per user. Open to feedback on the approach of adding user id to the key here, or in a separate PR.